### ISSUE TITLE
FBNeo core keeps crashing on PSVITA since stable 1.16.0.

### ISSUE THREAD

**Issue by jjkkyu** [libretro/RetroArch#16047]
## Description

FBNeo core keeps crashing on PSVITA since stable 1.16.0.
It used to work on 1.15.0

### Steps to reproduce the bug

Just load the FBNeo core and the app crashes.

### Version/Commit
You can find this information under Information/System Information

- RetroArch: 1.16.0

### Environment information

- OS: PSVITA1000 3.65 HENKAKU

**Comment by spriteice** [comment]
Happening for me too. Clean install of RA does not fix the issue.
Rolled back to 1.15.0 for now.

RetroArch Version 1.17.0

PSVITA2000 3.60 HENKAKU ENSO

**Comment by SerdarSaglam** [comment]
Fbneo keeps crashing, I haven't managed to run any games yet. Is there any chance we just get the old FBneo core?

RetroArch Version 1.17.0
OS: PSVITA1000 3.65 HENKAKU

**Comment by SerdarSaglam** [comment]
Can you share the 'fbneo_libretro.self' file included in Retroarch 1.15 version? 

This file is located in:   ux0:app/RETROVITA

**Comment by spriteice** [comment]
You can download the 1.15.0 VPK and extract it yourself.

I have done some further testing with 1.17.0 on my ROG Ally and Nintendo Switch and there are definitely some issues with the new FBneo core. Whilst it doesn’t crash outright like the Vita, Metal Slug 3 loads to a black screen on both of theses systems, although weirdly you can press insert coin and hear the noise. No other inputs do anything. This was never an issue before, and I have tried alternate roms. Rolling back to 1.15.0 or earlier on all my devices fixed the issue with FBneo. I went as far as 1.8.9 on my Vita as it’s far more stable than any of the newer versions.

**Comment by SerdarSaglam** [comment]
Yes man, I didn't know it could be opened with Winrar and it did :)

It works smoothly with kernel 1.17 over 1.15. For a much better experience, reduce the CPU speed to 50% in Fbneo core settings. Games are fluent and sound problems are resolved.

I just tested it, no more crashes and everything is smooth.

Download fbneo 1.15 core: https://github.com/SerdarSaglam/PsVita-RetroArch-Stable-Cores/blob/main/fbneo_libretro.self

ux0:app/RETROVITA  paste

**Comment by MrFriedRice** [comment]
1.16-1.17 ns switch and Android also crashed
1.15ok

**Comment by MrFriedRice** [comment]
> Can you try the fba2012 cores maybe? It might be that current FBNeo has too high RAM requirements for PS Vita nowadays. This can be dependent on the type of game you're trying to run on it.
#16353
Can this be fixed

**Comment by dfilskov** [comment]
> Can you try the fba2012 cores maybe? It might be that current FBNeo has too high RAM requirements for PS Vita nowadays. This can be dependent on the type of game you're trying to run on it.

FBA2012 works.

**Comment by dfilskov** [comment]
Snes9x 2010 keeps crashing too.

Snex9x 2005 Plus works, though.

**Comment by MrFriedRice** [comment]
> Snes9x 2010 v1.19.0 keeps crashing too.
> 
> ![snes_small2](https://private-user-images.githubusercontent.com/15096058/338605707-0a6876af-30b2-4178-b59e-8f92bdb07490.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTg2ODI4OTQsIm5iZiI6MTcxODY4MjU5NCwicGF0aCI6Ii8xNTA5NjA1OC8zMzg2MDU3MDctMGE2ODc2YWYtMzBiMi00MTc4LWI1OWUtOGY5MmJkYjA3NDkwLmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjE4VDAzNDk1NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY3ZmJmYWYzMjg4MmRmYTk2OWUzYjZlYjA4YjUxZjJiY2NmZmFkODBmNWE2OTdmOTY4ZjAyOTJlYWI1MWQ1NTkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0._Un-CviFWx9rbyII_1oEh_K9ndZPbcKh_u5ISt2u2bA)
> 
> (Snes9x 2005 Plus works, though)

Stop replying, they're not going to fix it anyway.

**Comment by TrellBell** [comment]
> > Can you try the fba2012 cores maybe? It might be that current FBNeo has too high RAM requirements for PS Vita nowadays. This can be dependent on the type of game you're trying to run on it.
> 
> FBA2012 works.
> 
> But when you've run the FBNeo once RA starts up using that core and crashes again making it impossible to oprn afterwards.
> 
> Is there a way to make RA start up without a core?

Go into the Retroarch folder in your data folder and delete salamander.cfg file you see. It creates it during the error and will crash upon opening Retroarch until it's deleted. Don't know how to get the core working though

**Comment by TrellBell** [comment]
> Yes man, I didn't know it could be opened with Winrar and it did :)
> 
> It works smoothly with kernel 1.17 over 1.15. For a much better experience in some games, reduce the CPU clock to 50% in Fbneo core settings. Games run smoothly and sounds are not distorted.
> 
> Of course, you need to find a different CPU clock setting for each game.
> 
> I just tested it, no more crashes and everything is smooth.
> 
> Download fbneo Retroarch 1.15 core: https://github.com/SerdarSaglam/PsVita-RetroArch-Stable-Cores/blob/main/fbneo_libretro.self
> 
> ux0:app/RETROVITA paste

This fixed the crashing issue. Thank you!

**Comment by barbudreadmon** [comment]
FBNeo dev here, core-specific problems should be reported to the team writing that core.

Is this still an issue ? Or does deleting that "salamander.cfg" file permanently fix that problem ? From what i'm reading here i find it very likely that this is the same kind of "core size limitation" that has been reached on several other console platforms (3ds, wiiu, ps3).

@spriteice should read the documentation : https://docs.libretro.com/library/fbneo/#why-do-i-get-a-black-screen-andor-cant-i-change-bios-in-neogeo-games

**Comment by dfilskov** [comment]
> FBNeo dev here, core-specific problems should be reported to the team writing that core.

Where can we find the repository for each core?

**Comment by frostiex** [comment]
I was able to get the games running, that's it. Running. As soon as I press any buttons. it crashes back to retroarch home screen. Is there a config file somewhere for FBneo?

**Comment by barbudreadmon** [comment]
@frostiex Make sure you are using the latest nightlies and not the retroarch 1.20 release.
It had a bug affecting all static platforms (switch, psvita, 3ds) : https://github.com/libretro/RetroArch/issues/17350
I asked @LibretroAdmin if a hotfix release could be made after fixing the bug but as far as i can tell there was no such release.

If that's not the problem, provide a log.
